x-pack/plugins/apm/public/application/index.tsx

```diff
@@ -5,99 +5,18 @@
 import { ConfigSchema } from '../';
 import { AppMountParameters, CoreStart } from '../../../../../src/core/public';
-import {
-  KibanaContextProvider,
-  RedirectAppLinks,
-  useUiSetting$,
-} from '../../../../../src/plugins/kibana_react/public';
-import { routes } from '../components/app/Main/route_config';
-import { ScrollToTopOnPathChange } from '../components/app/Main/ScrollToTopOnPathChange';
-import {
-  ApmPluginContext,
-  ApmPluginContextValue,
-} from '../context/apm_plugin/apm_plugin_context';
-import { LicenseProvider } from '../context/license/license_context';
-import { UrlParamsProvider } from '../context/url_params_context/url_params_context';
-import { useBreadcrumbs } from '../hooks/use_breadcrumbs';
 import { ApmPluginSetupDeps, ApmPluginStartDeps } from '../plugin';
 import { createCallApmApi } from '../services/rest/createCallApmApi';
 import { createStaticIndexPattern } from '../services/rest/index_pattern';
 import { setHelpExtension } from '../setHelpExtension';
 import { setReadonlyBadge } from '../updateBadge';
-import { AnomalyDetectionJobsContextProvider } from '../context/anomaly_detection_jobs/anomaly_detection_jobs_context';
-
-const MainContainer = euiStyled.div`
-  height: 100%;
-`;
-
-function App() {
-  const [darkMode] = useUiSetting$<boolean>('theme:darkMode');
-
-  useBreadcrumbs(routes);
-
-  return (
-    <ThemeProvider
-      theme={(outerTheme?: DefaultTheme) => ({
-        ...outerTheme,
-        eui: darkMode ? euiDarkVars : euiLightVars,
-        darkMode,
-      })}
-    >
-      <MainContainer data-test-subj="apmMainContainer" role="main">
-        <Route component={ScrollToTopOnPathChange} />
-        <Switch>
-          {routes.map((route, i) => (
-            <ApmRoute key={i} {...route} />
-          ))}
-        </Switch>
-      </MainContainer>
-    </ThemeProvider>
-  );
-}
-
-export function ApmAppRoot({
-  apmPluginContextValue,
-  startDeps,
-}: {
-  apmPluginContextValue: ApmPluginContextValue;
-  startDeps: ApmPluginStartDeps;
-}) {
-  const { appMountParameters, core } = apmPluginContextValue;
-  const { history } = appMountParameters;
-  const i18nCore = core.i18n;
-
-  return (
-    <RedirectAppLinks application={core.application}>
-      <ApmPluginContext.Provider value={apmPluginContextValue}>
-        <KibanaContextProvider services={{ ...core, ...startDeps }}>
-          <i18nCore.Context>
-            <Router history={history}>
-              <UrlParamsProvider>
-                <LicenseProvider>
-                  <AnomalyDetectionJobsContextProvider>
-                    <App />
-                  </AnomalyDetectionJobsContextProvider>
-                </LicenseProvider>
-              </UrlParamsProvider>
-            </Router>
-          </i18nCore.Context>
-        </KibanaContextProvider>
-      </ApmPluginContext.Provider>
-    </RedirectAppLinks>
-  );
-}
+import { ApmAppRoot } from '../components/routing/app_root';
```

Comment:
Moved all of this to the routing section (which contains routing and templating)